app/models/product_import/entry_validator.rb

```diff
@@ -73,11 +73,13 @@ def mark_as_new_variant(entry, product_id)
                                            'variant_unit_scale', 'primary_taxon_id')
       )
       new_variant.save
-      if entry.attributes['on_demand'].present?
-        new_variant.on_demand = entry.attributes['on_demand']
-      end
-      if entry.attributes['on_hand'].present?
-        new_variant.on_hand = entry.attributes['on_hand']
+      if entry.attributes['unit_type'].present? || entry.attributes['variant_unit_name'].present?
+        if entry.attributes['on_demand'].present?
+          new_variant.on_demand = entry.attributes['on_demand']
+        end
+        if entry.attributes['on_hand'].present?
+          new_variant.on_hand = entry.attributes['on_hand']
+        end
```

Comment:
Thanks for finding a solution. As you said yourself, it's not ideal. It's not obvious what the connection between the missing values and the stock information is.

The error message said that the variant needs to be saved. So maybe this is a better way?

```suggestion
      if new_variant.persisted?
        if entry.attributes['on_demand'].present?
          new_variant.on_demand = entry.attributes['on_demand']
        end
        if entry.attributes['on_hand'].present?
          new_variant.on_hand = entry.attributes['on_hand']
        end
```

It would also be great to write a spec to cover this case. I would have run that to test this code.

Reply:
Yes, it also works. I didn't pay due attention to the response of validate_date. So, it looks like a better way to solve the issue and works in other scenarios as well.

![Screen Shot 2022-01-29 at 11 39 10 (2)](https://user-images.githubusercontent.com/60043698/151650822-f50536f6-1e1b-4582-9ffd-def41792d2ad.png)